### ISSUE TITLE
mail/postfix: add settings for recipient address verification to GUI

### ISSUE THREAD

**Issue by WordsOfMe** [opnsense/plugins#2680]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md

[X] I have searched the existing issues and I'm convinced that mine is new.

[X] When the request is meant for an existing plugin, I've added its name to the title.

**Is your feature request related to a problem? Please describe.**
The following settings cannot be configured via GUI (here given with example values):
```
unverified_recipient_reject_code = 550
unverified_recipient_reject_reason = Address lookup failed
```

**Describe the solution you'd like**
Make the above settings configureable via GUI.

**Describe alternatives you've considered**
Manual editing of the postfix configuration is not permanent.

**Additional context**
- Recipient address verification will aim to clarify the existence of a mailbox at a downstream mail server, e.g. within a DMZ. The standard setting for `unverified_recipient_reject_code = 450` only generates a temporary failure. The sender will try to deliver the message again and again until it expires. This does not only increase the load on both OPNsense and the mail server, it also prevents the original sender from being informed via a bounce of the delivery problems in due time. Once everything is set up properly, the admin would usually like to set `unverified_recipient_reject_code = 550`.
- Without an additional setting postfix is quite chatty reporting a delivery problem caused by recipient address verification. The reply to the sending mail server looks like this:
`550 5.1.1 <john.doe@example.com>: Recipient address rejected: undeliverable address: host apollo.dmz.example.com [192.168.1.100] said: 550 5.1.1 <john.doe@example.com>: Recipient address rejected: example.com`
This discloses internal information about the topology (hostnames and IP) of the DMZ.

[postfix address verification](http://www.postfix.org/ADDRESS_VERIFICATION_README.html)

**Comment by JSuenram** [comment]
Is this feature implemented or not?

**Comment by mimugmail** [comment]
No, do you need it? Should be easy

**Comment by JSuenram** [comment]
> No, do you need it? Should be easy

Yeah, coming from Proxmox Mail Gateway where this is an configurable GUI-Option.

Idea is to have the opnsense check during connection if an rcpt to is valid by connecting in parallel to the backend SMTP-Server. Exchange in this case. Postfix itself can do this, but there are no options to enable by opnsense WebUI.

**Comment by mimugmail** [comment]
But are the two options mentioned above sufficient for this? I dont think so

**Comment by JSuenram** [comment]
> But are the two options mentioned above sufficient for this? I dont think so

As far as I know, Postfix can exactly do this by enabling the Adress Verification.

`A Postfix MTA verifies a sender or recipient address by probing the preferred MTAs for that address, without actually delivering mail. The preferred MTAs could include the Postfix MTA itself, or some remote MTAs (SMTP interruptus). Probe messages are like normal mail, except that they are never delivered, deferred or bounced; probe messages are always discarded.`

`Controlling the routing of address verification probes

By default, Postfix sends address verification probe messages via the same route as regular mail, because that normally produces the most accurate result. It's no good to verify a local address by connecting to your own SMTP port; that just triggers all kinds of mailer loop alarms. The same is true for any destination that your machine is best MX host for: hidden domains, virtual domains, etc.`


So, if you have a relay where you forward to set, it should verify if an address exists against this relay.

**Comment by WordsOfMe** [comment]
> Is this feature implemented or not?

This depends what you understand by "this feature".
You can enable recipent address verification in the OPNsense GUI, but the change I proposed is not implemented. This leads to the following behaviour:
- If a recipient address verification fails, e.g. because the user is unknown at the downstream mail server, OPNsense will only issue a temporary error (45x). A properly configured sending MTA will try to deliver again and again until the set threshold is reached.
- This temporary error will disclose the hostname and IP address of your downstream mail server, and this means leaking information about the internal topology of your network.

**Comment by JSuenram** [comment]
> > Is this feature implemented or not?
> 
> This depends what you understand by "this feature". You can enable recipent address verification in the OPNsense GUI, but the change I proposed is not implemented. This leads to the following behaviour:
> 
> * If a recipient address verification fails, e.g. because the user is unknown at the downstream mail server, OPNsense will only issue a temporary error (45x). A properly configured sending MTA will try to deliver again and again until the set threshold is reached.
> * This temporary error will disclose the hostname and IP address of your downstream mail server, and this means leaking information about the internal topology of your network.

Ok, I understand. In PMG you have the option to change the behavior. 

But in general it is working already if "Use Recipient Address Verification. Please keep in mind that this could put significant load onto the next server." is enabled in opnsense...


<img width="308" alt="image" src="https://github.com/opnsense/plugins/assets/68963439/cec0570b-dd92-44e5-9f7b-71ff999d6adc">
